text/0001-protocol-messages/0001-protocol-messages.md

```diff
@@ -0,0 +1,133 @@
+| ------ | ------------ |
+|  1     | Handshake    |
+|  2     | Legacy Tx Gossip |
+|  3     | Milestone Request|
+|  4     | Transaction Broadcast |
+|  5     | Transaction Request   |
+|  6     | Heartbeats            |
+
+# Motivation
+
+***The network rewrite change***
+1. Ability to kick-out a compromised/malfunctioning neighbor while preserving overall architecture performance. This means that a malfunctioning neighbor will only affect its own relationship with the node and will not affect how then node  treats other neighbors
+
+2. Drop support for UDP which is in practice slow (due to lack of congestion control) and not so reliable. Only utilize TCP.
+
+3. Enabling possibility to enforce global and per neighbor rate limits in both ingress and egress direction. 
+
+5. Enabling prioritization of message types.
+
+6. Enabling the addition of new message types and versioning the protocol. Thus paving the way for STING.
+
+
+***STING (Still TrInary Network Gossip)***
+
+1. Allow for faster syncing by
+    a. Seperating between requests and broadcasts of transactions.
+    b. Allowing to request specific milestones by index.
+    c. Sharing between nodes information on the milestones in their databases via Heartbeats
+    
+2. Eliminates fragmentation of request messages.
+
+
+
+
+# Detailed design
+
+## **TLV Messages - Protocol Version 1**
+The network rewrite introduces a new breaking protocol between nodes which works with type-length-value (TLV) denoted messages. Meaning that each sent message is composed of a header follwed by the message itself.
+
+| Order | Name   | Length (byte) | Desc             |
+| ----- | ----   | ------------- | ----                         |
+|   1   | Header |   3           | Metadata of message          |
+|   2   |Message |   Var         | Message itself               |
+
+#### Header
+Each message in the protocol is denoted by a 3 byte header:
+
+| Order | Name | Length (byte) | Desc             |
+| ----- | ---- | ------------- | ----                         |
+|   1   | Type |   1           | Type of message              |
+|   2   |Length|   2           | Length of message (65 KB max)|
+
+#### Handshake
+The handshake message is the first message which must be sent and received to/from a neighbor. It is used to construct the identity of the neighbor. If the advertised server socket port, coordinator address or mwm does not correspond to the receiving node’s configuration, the connection is dropped. It also sends its support for message types as a bitmap. Each index of the bitmap corresponds to a message type. If the bit on that index is turned on, then the corresponding message type is supported by the node. For example, `[01101110, 01010001]` denotes that this node supports message types 2, 3, 4, 6, 7, 9, 13 and 15. Thus, the length of the bitmap depends on the number of message types supported. The nodes can use that information to know what message types can be realyed to the peers.
```

Comment:
The explanation of the supported versions is wrong. It's not about message types but versions.